src/reporters/json.ts

```diff
@@ -219,6 +256,81 @@ function stepInfo(
   };
 }
 
+async function processScreenshot(screenshot: Buffer) {
+  const img = sharp(screenshot, { sequentialRead: true });
+  const { width, height } = await img.metadata();
+  /**
+   * Chop the screenshot image (1280*720) which is the default
+   * viewport size in to 64 equal blocks for a given image
+   * which can be acheived by keeping the division to 8
+   *
+   * Changing division to 16 would yield us 256 smaller blocks, but it is not
+   * optimal for caching in the ES and proved out to be bad, so we stick to 8
+   */
+  const divisions = 8;
+  const blockWidth = width / divisions;
+  const blockHeight = height / divisions;
+  const reference: ScreenshotReference = {
+    width,
+    height,
+    blockWidth,
+    blockHeight,
+    blocks: [],
+  };
+  const blocks: Array<ScreenshotBlob> = [];
+
+  for (let row = 0; row < divisions; row++) {
+    const top = row * blockHeight;
+    for (let col = 0; col < divisions; col++) {
+      const left = col * blockWidth;
+      const buf = await img
+        .extract({ top, left, width: blockWidth, height: blockHeight })
+        .webp({ quality: 80 })
+        .toBuffer();
+
+      const hash = createHash('sha1').update(buf).digest('hex');
```

Comment:
Are we certain that creating the webP image is deterministic? I'm wondering if there's any metadata encoded in there that would throw the hash off. It feels like it might be safer to hash the raw pixel buffer. However that might be slower because we might need to extract that buffer before encoding.

I think the way we can defend against this is to have a test where we validate that encoding the same screenshot twice yields identical blocks with identical hashes. It looks like the default behavior is no metadata, but it'd be good to confirm this: https://sharp.pixelplumbing.com/api-output#withmetadata

Reply:
Good point, adding a test for the same.

Reply:
Added the test, metadata is not added by default and test confirms it.